rectools/models/nn/transformers/torch_backbone.py

```diff
@@ -168,11 +179,34 @@ def forward(
         ----------
         sessions : torch.Tensor
             User sessions in the form of sequences of items ids.
+        TODO
 
         Returns
         -------
-        (torch.Tensor, torch.Tensor)
+        (torch.Tensor, torch.Tensor, torch.Tensor)
         """
         item_embs = self.item_model.get_all_embeddings()  # [n_items + n_item_extra_tokens, n_factors]
         session_embs = self.encode_sessions(sessions, item_embs)  # [batch_size, session_max_len, n_factors]
-        return item_embs, session_embs
+        logits = self.similarity_module(session_embs, item_embs, item_ids)
+        if last_n_items:
```

Comment:
I know I was the one to propose it, but could we please make a `forward` without an if? Let `forward` return all embeddings. And let's make a separate method that calls `forward` and returns only `-last_n_items` logits.
We are starting to think about ONNX and we now try to avoid `if` in `TorchBackbone`

Reply:
Also if we do the refactoring for `if` logic in calculating losses like I proposed below, we can skip implementing `-last_n_items` here. Because it will be quite easy to do in lightning module after refactoring.

Reply:
Fixed